pkg/services/accesscontrol/database/externalservices.go

```diff
@@ -227,6 +231,16 @@ func (*AccessControlStore) savePermissions(ctx context.Context, sess *db.Session
 			return errors.New("failed to delete permissions that have been removed from role")
 		}
 	}
+	if len(added) > 0 {
```

Comment:
I needed to swap the insertion and the removal to avoid conflicts when re-inserting permissions that did not have a scope split.